lib/facebook/messenger/server.rb

```diff
@@ -36,6 +37,8 @@ def verify_token
       end
 
       def receive
+        return if app_secret && !check_integrity
```

Comment:
If integrity check fails, the endpoint returns 200. Not sure if this is ok, but I think the integrity check will only fail if a request doesn't come from Facebook. If you return 50x, Facebook will retry the request many times.

Also, It only does integrity check if `app_secret` is present.